### PR TITLE
doc: Add guide for dev setup on Windows

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This adds a detailed instruction on how to get a working development setup on Windows.

State

- [x] Installation of WSL/Ubuntu
- [x] Cloning RIOT + first steps
- [x] Installation of VS Code
- [x] Integration of VS Code to WSL + IntelliSense
- [x] Working and compiling from VS Code
- [x] Working on real hardware via USB IP

### Testing procedure

[Check here](https://github.com/maribu/RIOT/tree/doc/getting-started/windows/doc/guides/setup-windows) for a rendering of the added README.md

### Issues/PRs references

https://github.com/RIOT-OS/RIOT/issues/20388

### CODE REVIEW

**Comment by dylad** [comment]
@maribu Note that you can also use .exe directly from WSL so one can install JLinkExe.exe (for example) into windows, symlink the .exe binary to JLinkExe within WSL then use good old `make flash` from WSL console to flash a board. This could be an interesting alternative to USBIP.
Of course, it will only work if the tool has a Windows version of it.

**Comment by maribu** [comment]
With USB-IP also `make term` works and it is actually quite straight forward. I just need to prepare the screen shots

**Comment by dylad** [comment]
Let me know when the full tutorial is ready. I'll give it a try on my W10 laptop.

**Comment by maribu** [comment]
Should be good now :)

**Comment by Teufelchen1** [comment]
Awesome! Please be patient for a couple of days, maybe one or two weeks. I'm drowning in work rn.

**Comment by dylad** [comment]
Followed the whole tutorial. I was able to successfully flash `p-nucleo-wb55` and `nrf5340dk-app` on W10 with WSL2.

The only issue is with DAP programmer which rely on USB HID. This is not supported by the default WSL Linux Kernel.
For now, I suggest to add a note in the tutorial to warn users.
We can add a specific workaround (using edbg windows binary) in a followup PR. Maybe this will be fix in a future WSL release.

For now, I think this PR is already in good shape !

**Comment by maribu** [comment]
Thanks. I added the info about USB HID programmers.

I do get why Microsoft disabled `CONFIG_USB_HIDDEV`, as users are not expected to forward their Mouse and Keyboard directly to WSL.

The thing is that HID devices are the only USB device class that works on Windows without any driver being installed. (Which also makes sense: Installing your Mouse and Keyboard driver without using a Mouse and Keyboard on Windows will be a challenge.)

Abusing HID as transport for debuggers only became popular because of the desire to have the smooth driver-less experience Linux users are familiar with also on Windows. It's quite ironic that this decision resulted in the experience now being less smooth for Windows users wanting to use RIOT.

File: doc/guides/setup-windows/README.md [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+3. You can drop out of the RIOT serial by pressing `Ctrl` + `C` and return
+   to the Linux shell.
+
+## Known Issues
+
+### Flashing Fails with Programmers using HID
+
+The Linux Kernel in WSL currently has
+[`CONFIG_USB_HIDDEV` disabled][wsl-hid-issue]. Hence, programmers using HID
+as transport do not work for know. The (non-conclusive) list of affected
```
Comment (dylad) [inline doc/guides/setup-windows/README.md:475]:
```suggestion
as transport do not work for now. The (non-conclusive) list of affected
```

**Review by dylad — CHANGES_REQUESTED** [review]
Spotted a typo in the newly added text.
You can squash directly.

**Review by dylad — APPROVED** [review]
Looking good !

**Comment by maribu** [comment]
Thx :)

**Pull request merged**